### PR TITLE
Queue members rework

### PR DESCRIPTION
**Pull request by Kaian**
#### Type Of Change 
- [ ] Small bug fix
- [X] New feature or enhancement
- [ ] Breaking change

#### Checklist:
- [X] Commits are named and have tag following [commit rules](https://github.com/irontec/ivozprovider/blob/bleeding/doc/dev/en/commits.md)
- [X] Commits are split per component (schema, web/admin, kamusers, agis, ..)
- [X] Changes have been tested locally
- [ ] Fixes an existing issue (Fixes #XXXX) 
- [ ] Upport from existing Pull request #XXXX

#### Description
This PR includes:
 - Conversion of asterisk queue members view into a table with required services to keep data updated
 - Changed Local Queue channels and agis to use extension instead of queue member id

Allowing:
 - Members can now store their pause status (not possible with a view)
 - Members are now updated properly in realtime queues whenever its data changes

### CODE REVIEW

File: library/Ivoz/Provider/Domain/Model/QueueMember/QueueMemberRepository.php [hunk 1]
```diff
@@ -7,4 +7,18 @@
...
+     */
+    public function findByUserId(int $userId): array;
+
+    /**
+     * @param int $queueid
+     * @param int $extension
+     * @return QueueMemberInterface|null
+     * @throws \Doctrine\ORM\NoResultException
+     * @throws \Doctrine\ORM\NonUniqueResultException
+     */
```
Comment (mmadariaga) [inline library/Ivoz/Provider/Domain/Model/QueueMember/QueueMemberRepository.php:22]:
```suggestion
    /**
     * @throws \Doctrine\ORM\NoResultException
     * @throws \Doctrine\ORM\NonUniqueResultException
     */
```

File: library/Ivoz/Provider/Domain/Model/QueueMember/QueueMemberRepository.php [hunk 2]
```diff
@@ -7,4 +7,18 @@
 
 interface QueueMemberRepository extends ObjectRepository, Selectable
 {
+    /**
+     * @param int $userId
+     * @return array<QueueMemberInterface>
+     */
```
Comment (mmadariaga) [inline library/Ivoz/Provider/Domain/Model/QueueMember/QueueMemberRepository.php:13]:
```suggestion
    /**
     * @return array<QueueMemberInterface>
     */
```

File: library/Ivoz/Ast/Infrastructure/Persistence/Doctrine/QueueMemberDoctrineRepository.php [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+{
+    public function __construct(ManagerRegistry $registry)
+    {
+        parent::__construct($registry, QueueMember::class);
+    }
+
+    /**
+     * @param int $queueMemberId
+     * @return QueueMemberInterface|null
+     */
```
Comment (mmadariaga) [inline library/Ivoz/Ast/Infrastructure/Persistence/Doctrine/QueueMemberDoctrineRepository.php:29]: Unnecessary

File: library/Ivoz/Ast/Domain/Model/QueueMember/QueueMemberRepository.php [hunk 4]
```diff
@@ -0,0 +1,11 @@
+<?php
+
+namespace Ivoz\Ast\Domain\Model\QueueMember;
+
+use Doctrine\Common\Collections\Selectable;
+use Doctrine\Persistence\ObjectRepository;
+
+interface QueueMemberRepository extends ObjectRepository, Selectable
+{
+    public function findOneByProviderQueueMemberId(int $queueMemberId): ?QueueMemberInterface;
```
Comment (mmadariaga) [inline library/Ivoz/Ast/Domain/Model/QueueMember/QueueMemberRepository.php:10]: Add ORM test for this method in schema/tests

File: library/Ivoz/Provider/Domain/Model/QueueMember/QueueMemberRepository.php [hunk 5]
```diff
@@ -7,4 +7,18 @@
...
+    public function findByUserId(int $userId): array;
```
Comment (mmadariaga) [inline library/Ivoz/Provider/Domain/Model/QueueMember/QueueMemberRepository.php:14]: Add tests in schema/tests

**Review by mmadariaga — CHANGES_REQUESTED** [review]

File: library/Ivoz/Ast/Domain/Service/QueueMember/UpdateByUser.php [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            self::EVENT_POST_PERSIST => self::POST_PERSIST_PRIORITY
+        ];
+    }
+
+    /**
+     * @return void
+     */
```
[the remark this answers is not in the document]
Comment (Kaian) [reply library/Ivoz/Ast/Domain/Service/QueueMember/UpdateByUser.php]: Fixed

File: library/Ivoz/Ast/Domain/Service/QueueMember/UpdateFromQueueMember.php [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+    public function execute(IvozQueueMemberInterface $ivozQueueMember): void
+    {
+        $queue = $ivozQueueMember->getQueue();
+        $user = $ivozQueueMember->getUser();
+        $terminal = $user->getTerminal();
+        $extension = $user->getExtension();
+
+        /** @var QueueMember|null $astQueueMember */
+        $astQueueMember = $this->queueMemberRepository->findOneByProviderQueueMemberId(
+            (int)$ivozQueueMember->getId()
```
[the remark this answers is not in the document]
Comment (Kaian) [reply library/Ivoz/Ast/Domain/Service/QueueMember/UpdateFromQueueMember.php]: Fixed

File: library/Ivoz/Ast/Infrastructure/Persistence/Doctrine/QueueMemberDoctrineRepository.php [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+    public function findOneByProviderQueueMemberId(int $queueMemberId): ?QueueMemberInterface
+    {
+        /** @var QueueMemberInterface $response */
```
[the remark this answers is not in the document]
Comment (Kaian) [reply library/Ivoz/Ast/Infrastructure/Persistence/Doctrine/QueueMemberDoctrineRepository.php]: Fixed

File: library/Ivoz/Provider/Domain/Model/QueueMember/QueueMemberRepository.php (hunk 1 above)
Comment (Kaian) [reply library/Ivoz/Provider/Domain/Model/QueueMember/QueueMemberRepository.php]: Fixed

File: library/Ivoz/Provider/Domain/Model/QueueMember/QueueMemberRepository.php (hunk 2 above)
Comment (Kaian) [reply library/Ivoz/Provider/Domain/Model/QueueMember/QueueMemberRepository.php]: Fixed

File: library/Ivoz/Provider/Domain/Model/QueueMember/QueueMemberRepository.php (hunk 5 above)
Comment (Kaian) [reply library/Ivoz/Provider/Domain/Model/QueueMember/QueueMemberRepository.php:13]: Added it_finds_by_user_id and it_finds_one_by_queue_and_extension repository tests

**Review by mmadariaga — APPROVED** [review]

**Pull request merged**